### ISSUE TITLE
False-positive for calling method of decorated class

### ISSUE THREAD

**Issue by xredor** [PyCQA/pylint#2941]
### Steps to reproduce
```
def class_instance(*args, **kwargs):
    def wrapper(cls):
        return cls(*args, **kwargs)
    return wrapper

@class_instance()
class A:
    def m(self, b):
        print(self, b)
        return id(self), id(b)

print(A.m(67) == (id(A), id(67)))
```

### Current behavior
Pylint raises an error `E: 12, 6: No value for argument 'b' in unbound method call (no-value-for-parameter)`

### Expected behavior
No error.

### pylint --version output
```
$ pylint3 --version
No config file found, using default configuration
pylint3 1.6.5, 
astroid 1.4.9
Python 3.5.3 (default, Sep 27 2018, 17:25:39) 
[GCC 6.3.0 20170516]
```

Debian stretch

**Comment by PCManticore** [comment]
Thanks for reporting an issue. Pylint currently does not understand decorators of this sort.
